serialisable.hpp

```diff
@@ -347,6 +378,48 @@ class Serialisable {
 		return true;
 	}
 
+	/*!
+	* \brief Saves or loads an arithmetic integer value
+	* \param The name of the value in the output/input file
+	* \param Reference to the value
+	* \return false if the value was absent while reading, true otherwise
+	*/
+	template<typename T>
+	typename std::enable_if<std::is_integral<T>::value, bool>::type
+		synch(const std::string& key, T& value) {
+		if (preferencesSaving_) {
+			preferencesJson_->getObject()[key] = std::make_shared<JSONint>(int64_t(value));
+		}
+		else {
+			auto found = preferencesJson_->getObject().find(key);
+			if (found != preferencesJson_->getObject().end()) {
+				value = T(found->second->getInt());
+			} return false;
+		}
+		return true;
+	}
+
+	/*!
+	* \brief Saves or loads an enum as integer
+	* \param The name of the value in the output/input file
+	* \param Reference to the value
+	* \return false if the value was absent while reading, true otherwise
+	*/
+	template<typename T>
+	typename std::enable_if<std::is_enum<T>::value, bool>::type
+		synch(const std::string& key, T& value) {
+		if (preferencesSaving_) {
+			preferencesJson_->getObject()[key] = std::make_shared<JSONint>(int(value));
```

Comment:
Please cast it to [`std::underlying_type_t<T>`](https://en.cppreference.com/w/cpp/types/underlying_type) instead of `int` for the case if someone uses a longer underlying type of enum (for bitmasking or something).

Reply:
Thank you for the correction, I will fix in the next commit.